src/app/core/states/auth/auth.effects.ts

```diff
@@ -0,0 +1,97 @@
+import { Injectable } from '@angular/core';
+import {
+  Actions,
+  createEffect,
+  ofType,
+  ROOT_EFFECTS_INIT,
+} from '@ngrx/effects';
+import { AuthenticationService } from '../../services/authentication.service';
+import { signIn, signUp, signOut, setToken } from './auth.actions';
+import { map, mergeMap, tap } from 'rxjs';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+
+@Injectable()
+export class AuthEffects {
+  constructor(
+    private actions$: Actions,
+    private authenticationService: AuthenticationService,
+    private router: Router,
+    private store: Store,
+  ) {}
+
+  signIn$ = createEffect(
+    () => {
+      return this.actions$.pipe(
+        ofType(signIn),
+        mergeMap(({ email, password }) => {
+          return this.authenticationService
+            .signInWithEmailPassword(email, password)
+            .pipe(
+              tap(() => {
+                this.router.navigate(['/']);
+              }),
+            );
+        }),
+      );
+    },
+    { dispatch: false },
+  );
+
+  signUp$ = createEffect(
+    () => {
+      return this.actions$.pipe(
+        ofType(signUp),
+        mergeMap(({ email, password, displayName }) => {
+          return this.authenticationService
+            .signUpWithEmailPassword(email, password, displayName)
+            .pipe(
+              tap(() => {
+                this.router.navigate(['/']);
+              }),
+            );
+        }),
+      );
+    },
+    { dispatch: false },
+  );
```

Comment:
Usually you would return an action that updates the store with the new state. The overall effect looks like you're using the action/effect infrastructure without the store? Is this mid-process or what's up with that?

Reply:
signIn and signUp now return a success function